### DISCUSSION TITLE
Should `CallIndirect` be in core hugr spec?

### DISCUSSION THREAD

**Discussion by cqc-alec** [Quantinuum/hugr#154] [Specification]
It's introduced in the spec as part of the "Higher-order (Tierkreis) Resource", rather than "Functional description", but is implemented as a core `OpType`. Where does it belong?

I think we were not intending to include higher-order functions in the core part of the spec; do we make an exception for "eval"?

**Comment by acl-cqc** [comment]
I think "eval" and `CallIndirect` are the same thing and that thing belongs in the higher-order/tierkreis resource

**Reply by cqc-alec** [reply]
Shall we convert this to an issue to remove `OpType::CallIndirect`? It isn't used anywhere yet as far as I can see.

**Reply by acl-cqc** [reply]
We should have that issue...but should we also include most of the `LeafOp`s? I thought we had an issue to make most of those into extension ops, but I don't see it - should we include those too? (@ss2165)

**Reply by ss2165** [reply]
yes I think leafop should mostly be operations for core spec types, the exact breakdown of which needs more thought first
